### PR TITLE
Patch to pass threescale info to next filter in the chain

### PR DESCRIPTION
**Pull request by rahulanand16nov**
This PR adds new configurable behavior to pass threescale's application to the next filter in the chain instead of doing an aurhrep call. Please suggest any edits that you think should be made before merging this PR.

### CODE REVIEW

**Comment by unleashed** [comment]
Can you take a look at the failed compilation error, @rahulanand16nov?

**Comment by rahulanand16nov** [comment]
> Can you take a look at the failed compilation error, @rahulanand16nov?

Passed 'cargo test'. Awaiting workflow approval.

File: src/proxy.rs
Comment (unleashed) [inline src/proxy.rs]: I'd move the `:` to only be added to the string if there is actually a key following it.

File: src/proxy.rs
Comment (unleashed) [inline src/proxy.rs]: This application type is legacy and only exists to support compatibility code that will never even remotely sit close to an Envoy/Proxy-Wasm gateway. In this case I'd just use return an error.

File: src/proxy.rs
Comment (unleashed) [inline src/proxy.rs]: Rather than creating a vector only to pick up more than one value, use a tuple of two values like this:

```rust
let (header, value) = match &apps[0] {
    // ...
    Application::UserKey(ref user_key) => ("x-3scale-user-key", user_key),
    // ...
}
```
and you will be able to `self.add_http_request_header(header, value);` afterwards.

File: src/configuration/api/v1.rs
Comment (unleashed) [inline src/configuration/api/v1.rs]: From the name I'm not sure what this means - how about `passthrough` or `passthrough_metadata` (even though today it cannot happen via metadata)?

**Review by unleashed — COMMENTED** [review]
Looks on the right path Rahul, but check out the comments.

Also, wrt style there are a few nits you can get fixed by following these steps:

1. When preparing a PR for review, or another round of reviews and the changes are very small, squash the commits so that there are no commits fixing errors introduced earlier in the same PR. When introducing large changes due to reviews, don't squash anything to ease the review, but do so before it is finally merged to have a clean commit history.
2. Split the commits so that you build on top of code/functions/structs introduced in previous commits - this is more an art than an exact science, but in this case first introduce the configuration knob and in a second commit add the logic to handle it.
3. Use a format for commit messages trying to follow the project's contributing guidelines and its previous commit and PR history, ie: in this case, use lowercase and a subject line with a length topping at around ~74 characters with a pattern like `subsystem-or-file: description in imperative tense`.

These rules are simple to follow once you get used to them and will help in your contributions everywhere (note: the rules are generic, but each project will have its own idiosyncrasies and require you jump over different hoops).

File: src/proxy.rs
Comment (rahulanand16nov) [reply src/proxy.rs]: Great idea! and... changed.

File: src/configuration/api/v1.rs
Comment (rahulanand16nov) [reply src/configuration/api/v1.rs]: `passthrough_metadata` sounds more right here, I guess? `passthrough` is more comparable to "just let the whole thing pass w/o doing anything".

File: src/proxy.rs
Comment (rahulanand16nov) [reply src/proxy.rs]: Right! We'll just have 1 item when splitting in the next filter if `app_key` is not available here.

**Review by unleashed — APPROVED** [review]

**Pull request merged**
